packages/edit-post/src/index.js

```diff
@@ -78,6 +79,19 @@ export function initializeEditor(
 	settings,
 	initialEdits
 ) {
+	// Prevent adding template part in the post editor.
+	// Only add the filter when the post editor is initialized, not imported.
+	addFilter(
+		'blockEditor.__unstableCanInsertBlockType',
+		'removeTemplatePartsFromInserter',
+		( can, blockType ) => {
+			if ( blockType.name === 'core/template-part' ) {
+				return false;
```

Comment:
We should check for the template mode here as well (the block should be allowed in template mode)